compiler/shared/main/scala/mlscript/compiler/ClassLifter.scala

```diff
@@ -0,0 +1,482 @@
+    def extT(rst: IterableOnce[TypeName]) = LocalContext(vSet, tSet -- rst)
+    def vSet2tSet = LocalContext(Set(), vSet.map(t => TypeName(t.name)))
+    def moveT2V(ts: Set[TypeName]) = {
+      val inters = tSet.intersect(ts)
+      LocalContext(vSet ++ inters.map(x => Var(x.name)), tSet -- inters)
+    }
+    def intersect(rst: LocalContext) = LocalContext(vSet intersect rst.vSet, tSet intersect rst.tSet)
+    override def toString(): String = "(" ++ vSet.mkString(", ") ++ "; " ++ tSet.mkString(", ") ++ ")"
+  }
+  private def asContext(v: Var) = LocalContext(Set(v), Set())
+  private def asContextV(vS: IterableOnce[Var]) = LocalContext(vS.toSet, Set())
+  private def asContext(t: TypeName) = LocalContext(Set(), Set(t))
+  private def asContextT(tS: IterableOnce[TypeName]) = LocalContext(Set(), tS.toSet)
+  private def emptyCtx = LocalContext(Set(), Set())
+
+  case class ClassInfoCache(
+      originNm: TypeName, 
+      liftedNm: TypeName, 
+      var capturedParams: LocalContext, 
+      fields: MSet[Var], 
+      innerClses: MMap[TypeName, ClassInfoCache], 
+      supClses: Set[TypeName],
+      outerCls: Option[ClassInfoCache],
+      body: NuTypeDef, 
+      depth: Int){
+        def ++(other: ClassInfoCache) = this.copy(
+            capturedParams = this.capturedParams ++ other.capturedParams, 
+            fields = this.fields ++ other.fields, 
+            innerClses = this.innerClses ++ other.innerClses
+          )
+        override def toString(): String = liftedNm.name ++ "@" ++ capturedParams.toString() ++ "^" ++ outerCls.map(_.liftedNm.toString()).getOrElse("_")
+      }
+  type ClassCache = Map[TypeName, ClassInfoCache]
+  type NamePath = List[String]
+
+  var retSeq: List[NuTypeDef] = Nil
+  var anonymCnt: Int = 0
+  var clsCnt: Int = 0
+  val logOutput: StringBuilder = new StringBuilder
+  val primiTypes = new mlscript.Typer(false, false, false).primitiveTypes
+
+  private def log(str: String): Unit = {
+    logOutput.append(str)
+    // println(str)
+  }
+  def getLog: String = logOutput.toString()
+
+  private def genAnoName(textNm: String = "Ano"): String = {
+    anonymCnt = anonymCnt + 1
+    textNm ++ "$" ++ anonymCnt.toString()
+  }
+
+  private def genParName(clsNm: String): String = "par$" ++ clsNm
+  
+  private def genInnerName(outerClsNm: TypeName, innerNm: String) = {
+    clsCnt = clsCnt+1
+    outerClsNm.name ++ "_" ++ innerNm ++ "$" ++ clsCnt.toString()
+  }
+
+  private def tupleEntityToVar(fld: (Option[Var], Fld)): Option[Var] = fld match{
+    case (None, Fld(_, _, v: Var)) => Some(v)
+    case (Some(v: Var), _) => Some(v)
+    case _ => None
+  }
+  private def tupleEntityMapTrm(f: Term => Term)(fld: (Option[Var], Fld)): (Option[Var], Fld) = fld match {
+    case (None, snd@Fld(_, _, trm)) => (None, snd.copy(value = f(trm)))
+    case _ => fld
+  }
+  private def tupleEntityRenameVar(f: Var => Var)(fld: (Option[Var], Fld)): (Option[Var], Fld) = fld match {
```

Comment:
This seems to be unused. Probably the same for the other similar functions.

Reply:
Sorry I didn’t receive notifications from Github so I missed these comments… this method only serves for previous version of the code and is indeed unused not. I will remove it in a later commit